### ISSUE TITLE
health_check_eviction_time_in_min is not applied to azurerm_linux_web_app in site_config block

### ISSUE THREAD

**Issue by samgriffiths11** [hashicorp/terraform-provider-azurerm#16302]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.8

### AzureRM Provider Version

3.1.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_web_app" "this" {
  name                = var.name
  location            = var.region
  resource_group_name = var.resource_group
  tags                = var.tags

  service_plan_id = azurerm_service_plan.this.id
  https_only      = true
  app_settings    = var.app_config

  auth_settings {
    enabled = false
  }

  identity {
    type = "SystemAssigned"
  }

  logs {
    application_logs {
      file_system_level = "Verbose"
    }

    http_logs {
      file_system {
        retention_in_days = 90
        retention_in_mb   = 100
      }
    }

    detailed_error_messages = true
    failed_request_tracing  = true
  }

  site_config {
    always_on                               = true
    ftps_state                              = "Disabled"
    health_check_path                       = var.health_check_path
    health_check_eviction_time_in_min       = 2
    container_registry_use_managed_identity = true
  }
}
```

### Debug Output/Panic Output

```shell
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # ...redacted.azurerm_linux_web_app.this will be updated in-place
  ~ resource "azurerm_linux_web_app" "this" {
      ...redacted

      ~ site_config {
          ~ health_check_eviction_time_in_min       = 0 -> 2
            # (24 unchanged attributes hidden)

            # (1 unchanged block hidden)
        }
        # (2 unchanged blocks hidden)
    }
```

### Expected Behaviour

The health_check_eviction_time_in_min change to 2 should be applied in the Azure Health Check "Configure the time that an unhealthy instance remains in the load balancer before it is removed." section for an Azure App Service (I1v2 App Service Plan).  

### Actual Behaviour

The value remains at 10 (default value):

![image](https://user-images.githubusercontent.com/87806583/162453688-a553f347-1a38-4287-be2e-0d5e13718dc5.png)

The state itself isn't updated (sensitive values redacted):

```
{
  "module": "..redacted",
  "mode": "managed",
  "type": "azurerm_linux_web_app",
  "name": "this",
  "provider": "provider[\"registry.terraform.io/hashicorp/azurerm\"]",
  "instances": [
    {
      "schema_version": 0,
      "attributes": {
        "app_settings": {
          ...redacted
        },
        "auth_settings": [
          {
            ...redacted
          }
        ],
        "backup": [],
        "client_affinity_enabled": false,
        "client_certificate_enabled": false,
        "client_certificate_mode": "Required",
        "connection_string": [],
        "custom_domain_verification_id": "...redacted",
        "default_hostname": "...redacted",
        "enabled": true,
        "https_only": true,
        "id": "...redacted",
        "identity": [
          {
            "identity_ids": [],
            "principal_id": "...redacted",
            "tenant_id": "...redacted",
            "type": "SystemAssigned"
          }
        ],
        "key_vault_reference_identity_id": "SystemAssigned",
        "kind": "app,linux,container",
        "location": "westeurope",
        "logs": [
          {
            "application_logs": [
              {
                "azure_blob_storage": [],
                "file_system_level": "Verbose"
              }
            ],
            "detailed_error_messages": true,
            "failed_request_tracing": true,
            "http_logs": [
              {
                "azure_blob_storage": [],
                "file_system": [
                  {
                    "retention_in_days": 90,
                    "retention_in_mb": 100
                  }
                ]
              }
            ]
          }
        ],
        "name": "...redacted",
        "outbound_ip_address_list": [
            ...redacted
        ],
        "outbound_ip_addresses": "...redacted",
        "possible_outbound_ip_address_list": [
            ...redacted
        ],
        "possible_outbound_ip_addresses": "...redacted",
        "resource_group_name": "...redacted",
        "service_plan_id": "...redacted",
        "site_config": [
          {
            "always_on": true,
            "api_definition_url": "",
            "api_management_api_id": "",
            "app_command_line": "",
            "application_stack": [
              {
                "docker_image": "...redacted",
                "docker_image_tag": "...redacted",
                "dotnet_version": "",
                "java_server": "",
                "java_server_version": "",
                "java_version": "",
                "node_version": "",
                "php_version": "",
                "python_version": "",
                "ruby_version": ""
              }
            ],
            "auto_heal_enabled": false,
            "auto_heal_setting": [],
            "container_registry_managed_identity_client_id": "",
            "container_registry_use_managed_identity": true,
            "cors": [],
            "default_documents": [
              "Default.htm",
              "Default.html",
              "Default.asp",
              "index.htm",
              "index.html",
              "iisstart.htm",
              "default.aspx",
              "index.php",
              "hostingstart.html"
            ],
            "detailed_error_logging_enabled": true,
            "ftps_state": "Disabled",
            "health_check_eviction_time_in_min": 0,
            "health_check_path": "/",
            "http2_enabled": false,
            "ip_restriction": [],
            "linux_fx_version": "...redacted",
            "load_balancing_mode": "LeastRequests",
            "local_mysql_enabled": false,
            "managed_pipeline_mode": "Integrated",
            "minimum_tls_version": "1.2",
            "remote_debugging_enabled": false,
            "remote_debugging_version": "VS2019",
            "scm_ip_restriction": [],
            "scm_minimum_tls_version": "1.2",
            "scm_type": "None",
            "scm_use_main_ip_restriction": false,
            "use_32_bit_worker": true,
            "vnet_route_all_enabled": false,
            "websockets_enabled": false,
            "worker_count": 1
          }
        ],
        "site_credential": [
          {
            "name": "$...redacted",
            "password": "...redacted"
          }
        ],
        "storage_account": [],
        "tags": {
          ...redacted
        },
        "timeouts": null
      },
      "sensitive_attributes": [
        ...redacted
      ],
      "private": "...redacted",
      "dependencies": [
        ...redacted
      ]
    }
  ]
}
```

### Steps to Reproduce

1. Set `site_config.health_check_eviction_time_in_min` to a value of `2` in the `azurerm_linux_web_app` resource
2. `terraform plan`
3. `terraform apply`

**Comment by toddlindsey** [comment]
And this isn't just with **azurerm_linux_web_app**, I see the same behavior here with **azurerm_windows_web_app** (I am also trying to change to 2, but it remains at 10)

**Comment by penfold** [comment]
Confirmed still issue in v3.19.1

**Comment by PrestonR** [comment]
Confirmed still issue in v3.28.0

**Comment by vgdomi** [comment]
Confirmed still issue in v3.28.0 but for azurerm_linux_web_app

**Comment by mggrpleg** [comment]
Confirmed still issue in v3.34 for azurerm_linux_web_app

**Comment by Mralbert93** [comment]
I don't believe the changes in [PR19685](https://github.com/hashicorp/terraform-provider-azurerm/pull/19685) which were released in 3.39 helped. It seems this is still an issue as we have customers reporting the same behavior in v3.42. Can we get someone to look into this? @jackofallops
